fix: Make sure the matched string is not simply a substring of a bigger word

Konubinix opened the pull request:
If one displayname is 'sam' and another's nickname is 'samy', then all events to samy were recognized as mentionning sam as well.

See the issue https://github.com/alphapapa/ement.el/issues/240

alphapapa commented:
Thanks. I generally prefer to use the `rx` macro for constructing regexps. If you're not familiar with it, the docstring explains it thoroughly, or I can make that change. 

Something that might be really helpful would be to think about scenarios in which this change might still not behave as expected or as intended, and write a few examples. That could help guide future development, and potentially save time by avoiding the retracing of steps. Or in other words, maybe we should write out a few test cases, even in plain English. 

Thanks for your help with this.

Konubinix committed:
fix: Make sure the matched string is not simply a substring of a bigger word

If one displayname is 'sam' and another's nickname is 'samy', then
all events to samy were recognized as mentionning sam as well.

See the issue https://github.com/alphapapa/ement.el/issues/240

Konubinix commented:
I used rx and added some test cases. I added a few one commented because I would have loved the code to deal with them properly but I could not find a regex that would work. Using the word-start and word-end is still a good way to get rid of a lot of false positives.

phil-s commented:
`(rx (regexp (regexp-quote foo)))` can just be `(rx (literal foo))`

Konubinix commented:
@phil-s thank you for the review. I took it into account in the code.

phil-s commented:
You might break the line here for formatting, but otherwise this PR looks good to me.

```diff
- (string-match-p (rx word-start (literal string) word-end) body))))
+ (string-match-p (rx word-start (literal string) word-end)
+                 body))))
```

Konubinix commented:
phil-s ***@***.***> writes:

> phil-s left a comment (alphapapa/ement.el#338)
>
> You might break the line here for formatting, but otherwise this PR looks good to me.
>
> ```diff
> - (string-match-p (rx word-start (literal string) word-end) body))))
> + (string-match-p (rx word-start (literal string) word-end)
> +                 body))))
> ```

Done

-- 
Konubinix
GPG Key    : 7439106A
Fingerprint: 5993 BE7A DA65 E2D9 06CE  5C36 75D2 3CED 7439 106A
